### PR TITLE
Switch to esbuild

### PR DESCRIPTION
**Pull request by kfcampbell**
Replicating [octokit/request.js#584](https://github.com/octokit/request.js/pull/584) to another repo. 

---

Pika has been deprecated for a while now, and the project has now been archived, and because of that the released npm package is missing most of the files generated by the build step.

Uses esbuild to transpile the TS source code into an ESM source, NodeJs bundle, and a browser bundle

Uses the TypeScript compiler to generate the types.

### Before the change?

* Uses pika for the build system
* The released npm package is missing most of the files generated by the build step. `dist-node`, `dist-types`, `dist-web`... even though they are generated correctly.

### After the change?

* Uses esbuild to generate the JS bundles
* Uses the TypeScript compiler to generate the type definitions
* The correct file matching patterns are used so we publish all the necessary files again.

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/main/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:

* N/A

### Pull request type

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
 // Import for types only, not a production dependency
 import { Octokit } from "@octokit/core";
```
Comment (wolfy1339) [inline src/index.ts:2]:
```suggestion
import type { Octokit } from "@octokit/core";
```

File: scripts/build.mjs [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import esbuild from "esbuild";
```
Comment (wolfy1339) [inline scripts/build.mjs:1]: It seems the indentation is wonky in this file

Comment (kfcampbell) [reply scripts/build.mjs:1]: `npm run lint` wasn't checking the `build.mjs` file. I've updated `package.json` accordingly and fixed the linting!

**Review by wolfy1339 — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -106,6 +89,6 @@
     ]
   },
   "engines": {
-    "node": ">= 14"
+    "node": ">= 18"
```
Comment (wolfy1339) [inline package.json:92]:
```suggestion
    "node": ">= 14"
```

File: scripts/build.mjs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  const entryPoints = ["./pkg/dist-src/index.js"];
+
+  await Promise.all([
+    // Build the a CJS Node.js bundle
+    esbuild.build({
+      entryPoints,
+      outdir: "pkg/dist-node",
+      bundle: true,
+      platform: "node",
+      target: "node18",
```
Comment (wolfy1339) [inline scripts/build.mjs:45]:
```suggestion
      target: "node14",
```

**Review by wolfy1339 — CHANGES_REQUESTED** [review]
To avoid this being a breaking change, let's not change these values

**Pull request merged**